dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java

```diff
@@ -0,0 +1,140 @@
+
+			// If idpConfig is null, means this site does not need SAML processing
+			if (identityProviderConfiguration != null && identityProviderConfiguration.isEnabled()) {
+
+				Logger.debug(this, () -> "Processing saml login request for idpConfig id: " + idpConfigId);
+				this.doRequestLoginSecurityLog(httpServletRequest, identityProviderConfiguration);
+
+				// This will redirect the user to the IdP Login Page.
+				this.samlAuthenticationService.authentication(httpServletRequest,
+						httpServletResponse, identityProviderConfiguration);
+
+				return Response.ok().build();
+			}
+		}
+
+		final String message = "No idpConfig for idpConfigId: " + idpConfigId + ". At " + httpServletRequest.getRequestURI();
+		Logger.debug( this, ()-> message);
+		throw new SamlException(message);
+	}
+
+	@GET
+	@Path( "/metadata/{idpConfigId}" )
+	@JSONP
+	@NoCache
+	@Produces( { MediaType.APPLICATION_JSON, "application/javascript" } )
+	// Gets metadata configuration by id
+	public void metadata( @PathParam( "idpConfigId" ) final String idpConfigId,
+						  @Context final HttpServletRequest httpServletRequest,
+						  @Context final HttpServletResponse httpServletResponse ) throws IOException {
+
+		boolean noConfig = true;
+
+		if (DotSamlFactory.getInstance().isAnyHostConfiguredAsSAML()) {
+
+			final IdentityProviderConfiguration identityProviderConfiguration =
+					this.identityProviderConfigurationFactory.findIdentityProviderConfigurationById(idpConfigId);
+
+			// If idpConfig is null, means this site does not need SAML processing
+			if (identityProviderConfiguration != null && identityProviderConfiguration.isEnabled()) {
+
+				Logger.debug(this, () -> "Processing saml login request for idpConfig id: " + idpConfigId);
+				this.samlAuthenticationService.renderMetadataXML(httpServletResponse.getWriter(), identityProviderConfiguration);
+				noConfig = false;
+			}
+		}
+
+		if (noConfig) {
+
+			final String message = "No idpConfig for idpConfigId: " + idpConfigId + ". At " + httpServletRequest.getRequestURI();
+			Logger.debug(this, () -> message);
+			throw new SamlException(message);
+		}
+	}
+
+
+	public void doRequestLoginSecurityLog(final HttpServletRequest request,
+										  final IdentityProviderConfiguration identityProviderConfiguration) {
+
+		try {
+
+			final Host host  = this.hostWebAPI.getCurrentHost(request);
+			final String env = this.isFrontEndLoginPage(request.getRequestURI()) ? "frontend" : "backend";
+			final String log = new Date() + ": SAML login request for Site '" + host.getHostname() + "' with IdP ID: "
+					+ identityProviderConfiguration.getId() + " (" + env + ") from " + request.getRemoteAddr();
+
+			// “$TIMEDATE: SAML login request for $host (frontend|backend)from
+			// $REQUEST_ADDR”
+			SecurityLogger.logInfo(SecurityLogger.class, this.getClass() + " - " + log);
+			Logger.debug(this, ()-> log);
+		} catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)